File: BinarizationAndDenoising/MedianFilter/MedianFilter.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(int argc, char **argv)
{
	Mat src = imread("D:/Images/saltimg.jpg");
	if (src.empty())
	{
		cout << "can not find the img" << endl;
		return -1;
	}
	namedWindow("src", CV_WINDOW_AUTOSIZE);
	imshow("src", src);

	Mat dst;
	//第三个参数表示统计中位数的窗口大小为5x5
	medianBlur(src, dst, 5);
	imshow("median blur", dst);
	
	waitKey(0);
	return 0;
}

File: BinarizationAndDenoising/MedianFilter/MedianFilter.py
import cv2 as cv
import numpy as np

src = cv.imread("D:/Images/saltimg.jpg")

dst = cv.medianBlur(src, 5)

height, width = src.shape[:2]
result = np.zeros([height, width * 2, 3], dtype=src.dtype)
result[0: height, 0: width, :] = src
result[:, width: width * 2, :] = dst
cv.putText(result, "salt_src", (10, 50), cv.FONT_HERSHEY_SIMPLEX, 2.0, (0, 255, 255), 1)
cv.putText(result, "median_img", (10 + width, 50), cv.FONT_HERSHEY_SIMPLEX, 2.0, (0, 255, 255), 1)
cv.imshow("result", result)

cv.waitKey(0)
cv.destroyAllWindows()
